<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1098</width>
    <height>1008</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gBox_1">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>1041</width>
      <height>681</height>
     </rect>
    </property>
    <property name="title">
     <string>Xử lý 1 ảnh</string>
    </property>
    <widget class="QLabel" name="lb_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>36</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Tên file ảnh</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>85</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ânh âm bản</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Mức sáng</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Độ tương phản</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>260</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Chọn ngưỡng</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_6">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>140</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>&gt; 0 tăng, &lt; 0 giảm</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_7">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>200</y>
       <width>221</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>&gt; 1 tăng 0 &lt; c &lt; 1 giảm</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_8">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>250</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>&gt; n trắng &lt; n đen</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_filename">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>631</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_mucsang">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>130</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_dotuongphan">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>190</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_phannguong">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>250</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="bt_Chon">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>30</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Chọn</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_amban_xam">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>80</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_amban_mau">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>80</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_mucsang_xam">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>130</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_mucsang_mau">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>130</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_dotuongphan_xam">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>190</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_dotuongphan_mau">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>180</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_phannguong">
     <property name="geometry">
      <rect>
       <x>620</x>
       <y>250</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Xử lý phân ngưỡng</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gaithuat_tuantumau">
     <property name="geometry">
      <rect>
       <x>850</x>
       <y>230</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Giải thuật tuần tự</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_giaithuat_tosu">
     <property name="geometry">
      <rect>
       <x>850</x>
       <y>280</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Giải thuật Otsu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_histogram_xam">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>310</y>
       <width>231</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Vẽ Histogram ảnh xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_histogram_mau">
     <property name="geometry">
      <rect>
       <x>580</x>
       <y>310</y>
       <width>221</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Vẽ Histogram ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_biendoituyentinh_histogram">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>360</y>
       <width>271</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Biến đổi tuyến tính Histogram</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_canbang_histogram">
     <property name="geometry">
      <rect>
       <x>560</x>
       <y>360</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Cân bằng Histogram</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_kichthuocmatna">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>420</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Kích thước mặt nạ</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_tilenhieu">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>470</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Tỷ lệ nhiễu (%)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_kickthuocmatna">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>420</y>
       <width>113</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_tylenhieu">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>470</y>
       <width>113</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_anmon">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>620</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Ăn mòn</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gianno">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>620</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Giỡn nở</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_open">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>620</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_close">
     <property name="geometry">
      <rect>
       <x>720</x>
       <y>620</y>
       <width>161</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Close</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loctrungbinh">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>420</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc trung bình </string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loctrungvi">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>420</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc trung vị</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gauss_laplape">
     <property name="geometry">
      <rect>
       <x>760</x>
       <y>420</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Gauss + Laplape</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_nhieumuoitieu">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>470</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Nhiễu muối tiêu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_themnhieu_anhmau">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>470</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Thêm nhiễu ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_g_l_anhmau">
     <property name="geometry">
      <rect>
       <x>760</x>
       <y>470</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>G+L Ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_gradient">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>520</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Gradient</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gradient_anhmau">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>570</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Gradient ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_robert">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>520</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Robert</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_robert_anhmau">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>570</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Robert ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_sobel">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>520</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Sobel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_sobel_anhmau">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>570</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Sobel ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_prewitt">
     <property name="geometry">
      <rect>
       <x>610</x>
       <y>520</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Prewitt</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_prewitt_anhmau">
     <property name="geometry">
      <rect>
       <x>610</x>
       <y>570</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Prewitt ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_laplace">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>520</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Laplace</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loc_laplace_anhmau">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>570</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc Laplace ảnh màu</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gBox_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>690</y>
      <width>1051</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>Xử lý 2 ảnh</string>
    </property>
    <widget class="QLineEdit" name="ln_anh1">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>30</y>
       <width>601</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_anh2">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>90</y>
       <width>601</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_Opacity1">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>150</y>
       <width>241</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ln_Opacity1_2">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>210</y>
       <width>241</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ChonAnh1">
     <property name="geometry">
      <rect>
       <x>780</x>
       <y>30</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Chọn ảnh 1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ChonAnh2">
     <property name="geometry">
      <rect>
       <x>780</x>
       <y>90</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Chọn ảnh 2</string>
     </property>
    </widget>
    <widget class="QPushButton" name="bt_Cong2Anh">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>180</y>
       <width>161</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Cộng 2 ảnh</string>
     </property>
    </widget>
    <widget class="QPushButton" name="bt_Tru2Anh">
     <property name="geometry">
      <rect>
       <x>610</x>
       <y>180</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Trừ 2 ảnh</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_Opacity1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Opacity ảnh 1</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_Opacity1_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Opacity ảnh 2</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1098</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
